srcts/src/components/card.ts

```diff
@@ -0,0 +1,277 @@
+    // full screen card, so we move focus to the close button.
+    if (!(event instanceof KeyboardEvent)) return;
+    if (!this.container.matches(`.${Card.attr.CLASS_FULL_SCREEN}`)) return;
+    if (event.key === "Tab") {
+      event.preventDefault();
+      this.overlay.anchor.focus();
+    }
+  }
+
+  private _createOverlay(): CardOverlay {
+    const container = document.createElement("div");
+    container.id = Card.attr.ID_FULL_SCREEN_OVERLAY;
+    container.classList.add(Card.attr.ID_FULL_SCREEN_OVERLAY);
+
+    const anchor = document.createElement("a");
+    anchor.classList.add(Card.attr.CLASS_FULL_SCREEN_EXIT);
+    anchor.tabIndex = 0;
+    anchor.onclick = () => this.exitFullScreen();
+    anchor.onkeyup = (ev) => {
+      if (ev.key === "Enter" || ev.key === " ") {
+        this.exitFullScreen();
+      }
+    };
+    anchor.onkeydown = (ev) => {
+      // if tabbing backwards out of the card,
+      // cycle focus back to last focus element within the card
+      if (ev.key === "Tab" && ev.shiftKey) {
+        ev.preventDefault();
+        this.lastFocusInterior?.focus();
+      }
+    };
+    anchor.innerHTML = this._overlayCloseHtml();
+
+    container.appendChild(anchor);
+    return { container, anchor };
+  }
+
+  private _overlayCloseHtml(): string {
+    return (
+      "Close " +
+      "<svg width:'20' height='20' fill='currentColor' class='bi bi-x-lg' " +
+      "viewBox='0 0 16 16'>" +
+      "<path d='M2.146 2.854a.5.5 0 1 1 .708-.708L8 7.293l5.146-5.147a.5.5 " +
+      "0 0 1 .708.708L8.707 8l5.147 5.146a.5.5 0 0 1-.708.708L8 8.707l-5.146 " +
+      "5.147a.5.5 0 0 1-.708-.708L7.293 8 2.146 2.854Z'/></svg>"
+    );
+  }
+
+  private static instanceMap: WeakMap<HTMLElement, Card> = new WeakMap();
+
+  public static getInstance(el: HTMLElement): Card | undefined {
+    return Card.instanceMap.get(el);
+  }
+
+  /**
+   * If cards are initialized before the DOM is ready, we re-schedule the
+   * initialization to occur on DOMContentLoaded.
+   * @private
+   * @static
+   * @type {boolean}
+   */
+  private static onReadyScheduled = false;
+
+  public static initializeAllCards(flushResizeObserver = true): void {
+    if (document.readyState === "loading") {
+      if (!Card.onReadyScheduled) {
+        Card.onReadyScheduled = true;
+        document.addEventListener("DOMContentLoaded", () => {
+          Card.initializeAllCards(false);
+        });
+      }
+      return;
+    }
+
+    if (flushResizeObserver) {
+      // Trigger a recheck of observed cards to unobserve non-existent cards
+      Card.shinyResizeObserver.flush();
+    }
+
+    const initSelector = `.${Card.attr.CLASS_CARD}[${Card.attr.ATTR_INIT}]`;
+    if (!document.querySelector(initSelector)) {
+      // no cards to initialize
+      return;
+    }
+
+    const cards = document.querySelectorAll(initSelector);
+    cards.forEach((card) => new Card(card as HTMLElement));
+  }
+}
```

Comment:
Would it be worth abstracting this out to avoid duplication of logic with sidebars?